### PR TITLE
add curl guide to docs

### PR DESCRIPTION
**Pull request by abemassry**

### CODE REVIEW

File: docs/src/components/guides/curl/CreateLinode.js [hunk 1]
```diff
@@ -0,0 +1,291 @@
...
+        /* and so on */
+    ],
+    "page": 1,
+    "total_pages": 1,
+    "total_results": 7
+}`}
+          </code>
+        </pre>
+        <p>
+          The region list is pretty self-explanatory: regions geographical locations are provided in the <code>label</code> field. The <code>id</code> field is a unique ID which you’ll use to refer to the region you want to select. For this example, we’ll go with the "Newark, NJ" region with ID "us-east-1a".
```
Comment (mparke) [inline docs/src/components/guides/curl/CreateLinode.js]: `region's`?

File: docs/src/components/guides/curl/CreateLinode.js [hunk 2]
```diff
@@ -0,0 +1,291 @@
...
+        /* and so on */
+    ],
+    "page": 1,
+    "total_pages": 1,
+    "total_results": 1
+}`}
+          </code>
+        </pre>
+        <p>
+          Each entry in the services list represents an available Linode service plan. You can review the details of each plan, such as the number of Virtual CPUs, memory, disk space, monthly price, etc. For explanations of each field, see the <Link to="/reference/endpoints/linode/types">complete service reference</Link>. Once you have selected the service plan you want to launch, note its ID and continue to the next step.
```
Comment (mparke) [inline docs/src/components/guides/curl/CreateLinode.js]: `Virtual` > `virtual` ?

File: docs/src/components/guides/curl/CreateLinode.js [hunk 3]
```diff
@@ -0,0 +1,291 @@
...
+        /* and so on */
+    ],
+    "page": 1,
+    "total_pages": 1,
+    "total_results": 1
+}`}
+          </code>
+        </pre>
+        <p>
+          For detailed information about each field, see the complete distribution reference. For this example, we’ll go with Debian 8.
```
Comment (mparke) [inline docs/src/components/guides/curl/CreateLinode.js]: link to distribution reference here?

File: docs/src/components/guides/curl/CreateLinode.js (hunk 1 above)
Comment (eatonphil) [inline docs/src/components/guides/curl/CreateLinode.js]: regions'?

File: docs/src/components/guides/curl/TestingWithCurl.js [hunk 4]
```diff
@@ -0,0 +1,171 @@
...
+export default function Introduction() {
+  return (
+    <section className="Article">
+      <header>
+          <h1>Testing with cURL</h1>
+          <h5>by William Smith; last updated May 22nd, 2017</h5>
+      </header>
+      <section>
+        <p>
+          <a href="http://curl.haxx.se/">cURL</a> is a simple and popular command line tool that allows you to perform various kinds of HTTP requests. It may already be installed on your system - run curl --version in your shell to check. Once you've confirmed that you have it installed, we can use it to test out the Linode API from the comfort of your shell.
```
Comment (eatonphil) [inline docs/src/components/guides/curl/TestingWithCurl.js]: `run curl --version` could we put the code part in `<code>` or whatever so you can read that it is a command?

**Comment by eatonphil** [comment]
There is inconsistent use of at least 2nd-person vs 1st-person. Could we pick one and stick with it?

File: docs/src/components/guides/curl/TestingWithCurl.js [hunk 5]
```diff
@@ -0,0 +1,171 @@
...
+        </p>
+      </section>
+      <section>
+        <h2>Unauthenticated Requests</h2>
+        <p>
+          You can perform anonymous HTTP requests against various resources on the API. You can tell which ones are anonymous from the <i className="fa fa-lock"></i> Authenticated indicator in the reference documentation. For API endpoints where this indicator is missing, you're able to use curl to test them without any additional steps. For example, we could <Link to="/reference/endpoints/linode/distributions">list supported distributions</Link>:
```
Comment (eatonphil) [inline docs/src/components/guides/curl/TestingWithCurl.js]: `<i className="fa fa-lock"></i> Authenticated` this is not currently a thing.

File: docs/src/components/guides/curl/TestingWithCurl.js [hunk 6]
```diff
@@ -0,0 +1,171 @@
...
+          </code>
+        </pre>
+      </section>
+      <section>
+        <h2>Authenticated Requests</h2>
+        <p>
+          For many requests, you will have to authenticate as a particular user. For now, we’re going to use a personal access token to make things easier. If you’d like to learn how to make a full blown OAuth client, read the <Link to="/access">authentication documentation</Link>.
+        </p>
+        <p>
+          To generate a personal access token <a href="https://login.alpha.linode.com/register/">sign up for the api alpha</a> and a token will be emailed to you. These tokens can be used to make API requests authenticated with your Linode account and have full access to all OAuth scopes. You’ll only see the full OAuth token once, so be sure to write it down somewhere. If you’re in the shell, running something like this might work well:
```
Comment (eatonphil) [inline docs/src/components/guides/curl/TestingWithCurl.js]: To generate a personal access token,

Comment (eatonphil) [inline docs/src/components/guides/curl/TestingWithCurl.js]: to make authenticated API requests

File: docs/src/components/guides/curl/TestingWithCurl.js [hunk 7]
```diff
@@ -0,0 +1,171 @@
...
+        <pre>
+          <code>
+{`token="that token"`}
+          </code>
+        </pre>
+      </section>
+      <section>
+        <h2>Authentication Header</h2>
+        <p>
+          Now you can make requests with curl using your access token by adding <code>-H "Authorization: token $token"</code>. The <i className="fa fa-lock"></i> Authenticated requests on the reference page include this header in the curl examples. Try this, for example:
```
Comment (eatonphil) [inline docs/src/components/guides/curl/TestingWithCurl.js]: s/Try this,//g

File: docs/src/components/guides/curl/CreateLinode.js [hunk 8]
```diff
@@ -0,0 +1,291 @@
...
+          Creating a Linode requires you to be logged in. Before proceeding, make sure you have gone through the <Link to="/guides/curl/testing-with-curl">testing with curl guide</Link>, since you will need an authorization token to proceed.
+        </p>
+        <p>
+          Before you can create a Linode, you will need to choose a <strong>region</strong>, a <strong>service plan</strong>, and a <strong>distribution</strong>. The API has endpoints available to help you do this.
+        </p>
+      </section>
+      <section>
+        <h2>Selecting a region</h2>
+        <p>
+          A region is a physical location which can run Linodes. To retrieve a list of available region, you can use the /regions API endpoint. To make an API call against this endpoint over curl, run the following command:
```
Comment (eatonphil) [inline docs/src/components/guides/curl/CreateLinode.js]: that contains Linodes.?

Comment (eatonphil) [inline docs/src/components/guides/curl/CreateLinode.js]: Link to /regions documentation?

File: docs/src/components/guides/curl/CreateLinode.js [hunk 9]
```diff
@@ -0,0 +1,291 @@
...
+            curl {API_ROOT}/{API_VERSION}/linode/types
+          </code>
+        </pre>
+        <p>
+          The above command will return a JSON object like the following:
+        </p>
+        <pre>
+          <code>
+{`{
+    "services": [
```
Comment (eatonphil) [inline docs/src/components/guides/curl/CreateLinode.js]: types

**Comment by eatonphil** [comment]
`s/(service|service plan)/type/g`?

File: docs/src/components/guides/curl/CreateLinode.js (hunk 3 above)
Comment (eatonphil) [inline docs/src/components/guides/curl/CreateLinode.js]: +1

File: docs/src/components/guides/curl/CreateLinode.js [hunk 10]
```diff
@@ -0,0 +1,291 @@
...
+          For detailed information about each field, see the complete distribution reference. For this example, we’ll go with Debian 8.
+        </p>
+      </section>
+      <section>
+        <h2>Creating your new Linode</h2>
+        <p>
+          Now that you’ve selected a region, service plan, and distribution, you’re ready to launch a new Linode! The API calls above were unauthenticated because they return only publicly visible information. However, launching a Linode is tied to your account so this call must be authenticated.
+        </p>
+        <p>
+          You will need to substitute your authorization token in the command below before running it (replace <strong>$TOKEN</strong>). If you don’t yet have an authorization token, read through the Testing with curl guide before proceeding.
```
Comment (eatonphil) [inline docs/src/components/guides/curl/CreateLinode.js]: Capitalization same as article title and link to article?

File: docs/src/components/guides/curl/CreateLinode.js [hunk 11]
```diff
@@ -0,0 +1,291 @@
...
+        <p>
+          Run the SSH command from the Linode response object and enter the root password you set in the create Linode step:
+        </p>
+        <pre>
+          <code>
+{`ssh root@$public_ip`}
+          </code>
+        </pre>
+        <p>
+          Congratulations! You have now successfully launched a Linode through the API!
```
Comment (eatonphil) [inline docs/src/components/guides/curl/CreateLinode.js]: s/API!/API./?

File: docs/src/components/guides/curl/CreateLinode.js (hunk 1 above)
Comment (mparke) [inline docs/src/components/guides/curl/CreateLinode.js]: 😅  this ^

File: docs/src/components/guides/curl/TestingWithCurl.js (hunk 5 above)
Comment (abemassry) [inline docs/src/components/guides/curl/TestingWithCurl.js]: Not sure how to reword this with leaving it out.

File: docs/src/components/guides/curl/CreateLinode.js (hunk 10 above)
Comment (abemassry) [inline docs/src/components/guides/curl/CreateLinode.js]: not sure which you're referring to

File: docs/src/components/guides/curl/TestingWithCurl.js [hunk 12]
```diff
@@ -0,0 +1,171 @@
...
+        <pre>
+          <code>
+{`token="that token"`}
+          </code>
+        </pre>
+      </section>
+      <section>
+        <h2>Authentication Header</h2>
+        <p>
+          Now you can make requests with curl using your access token by adding <code>-H "Authorization: token $token"</code>. The <i className="fa fa-lock"></i> Authenticated requests on the reference page include this header in the curl examples. For example:
```
Comment (mparke) [inline docs/src/components/guides/curl/TestingWithCurl.js]: needs to use constant var `LOGIN_ROOT`

File: docs/src/components/guides/curl/TestingWithCurl.js [hunk 13]
```diff
@@ -0,0 +1,171 @@
...
+export default function Introduction() {
+  return (
+    <section className="Article">
+      <header>
+          <h1>Testing with cURL</h1>
+          <h5>by William Smith; last updated May 22nd, 2017</h5>
+      </header>
+      <section>
+        <p>
+          <a href="http://curl.haxx.se/">cURL</a> is a simple and popular command line tool that allows you to perform various kinds of HTTP requests. It may already be installed on your system - run <code>curl --version</code> in your shell to check. Once you've confirmed that you have it installed, we can use it to test out the Linode API from the comfort of your shell.
```
Comment (eatonphil) [inline docs/src/components/guides/curl/TestingWithCurl.js]: https://github.com/linode/manager/blob/master/src/components/SessionMenu.js#L32

File: docs/src/components/guides/curl/TestingWithCurl.js [hunk 14]
```diff
@@ -0,0 +1,171 @@
...
+          </code>
+        </pre>
+      </section>
+      <section>
+        <h2>Authenticated Requests</h2>
+        <p>
+          For many requests, you will have to authenticate as a particular user. For now, we’re going to use a personal access token to make things easier. If you’d like to learn how to make a full blown OAuth client, read the <Link to="/access">authentication documentation</Link>.
+        </p>
+        <p>
+          To generate a personal access token, <a href="https://login.alpha.linode.com/register/">sign up for the api alpha</a> and a token will be emailed to you. These tokens can be used to make authenticated API requests with your Linode account and have full access to all OAuth scopes. You’ll only see the full OAuth token once, so be sure to write it down somewhere. If you’re in the shell, running something like this might work well:
```
Comment (mparke) [inline docs/src/components/guides/curl/TestingWithCurl.js]: url needs to use constant var `LOGIN_ROOT`, for example ```href={`${LOGIN_ROOT}/register`}```

File: docs/src/components/guides/curl/TestingWithCurl.js (hunk 13 above)
Comment (eatonphil) [inline docs/src/components/guides/curl/TestingWithCurl.js]: https://www.jitbit.com/alexblog/256-targetblank---the-most-underestimated-vulnerability-ever/

File: docs/src/components/guides/curl/TestingWithCurl.js (hunk 5 above)
Comment (mparke) [inline docs/src/components/guides/curl/TestingWithCurl.js]: url on this line will need `ROUTE_BASE_PATH` const templated

File: docs/src/components/guides/curl/Index.js [hunk 15]
```diff
@@ -0,0 +1,40 @@
...
+            headerClassName: 'SectionColumn',
+            hrefFn: function(subPage) {
+              return subPage.href;
+            }
+          },
+          { label: 'Description', dataKey: 'description' },
+        ]}
+        data={[
+          {
+            href: '/guides/curl/testing-with-curl',
```
Comment (mparke) [inline docs/src/components/guides/curl/Index.js]: template w/ ROUTE_BASE_PATH

File: docs/src/components/guides/curl/Index.js [hunk 16]
```diff
@@ -0,0 +1,40 @@
...
+            path: 'Testing with cURL',
+            description: 'A crash course on cURL and how to use it to try out the API.',
+          },
+          {
+            href: '/guides/curl/creating-a-linode',
```
Comment (mparke) [inline docs/src/components/guides/curl/Index.js]: template w/ ROUTE_BASE_PATH

File: docs/src/App.js [hunk 17]
```diff
@@ -85,6 +88,9 @@ export function init() {
           <Route path="/errors" component={Errors} />
           <Route path="/guides/python" component={Python} />
           <Route path="/guides/python/introduction" component={PythonIntroduction} />
+          <Route path="/guides/curl" component={Curl} />
```
Comment (mparke) [inline docs/src/App.js]: routes added will need `ROUTE_BASE_PATH`, see example lines above in the file after you rebase

File: docs/src/components/guides/curl/TestingWithCurl.js [hunk 18]
```diff
@@ -0,0 +1,175 @@
...
+            {
+               "backups_price": 250,
+               "label": "Linode 2048",
+               "storage": 24576,
+               "transfer": 2000,
+               "vcpus": 1,
+               "id": "g5-standard-1",
+               "hourly_price": 1,
+               "ram": 2048,
+               "monthly_price": 1000,
```
Comment (na3d) [inline docs/src/components/guides/curl/TestingWithCurl.js]: Price return has been fixed, so it should be:

> "monthly_price": 10.0,

File: docs/src/components/guides/curl/TestingWithCurl.js [hunk 19]
```diff
@@ -0,0 +1,175 @@
...
+               "threshold": 80
+            },
+            "cpu": {
+               "enabled": true,
+               "threshold": 90
+            }
+         },
+         "type": [
+            {
+               "backups_price": 250,
```
Comment (na3d) [inline docs/src/components/guides/curl/TestingWithCurl.js]: Like below, price has been fixed, so 2.5.  Should double check everywhere to make sure price is fixed.

Comment (na3d) [inline docs/src/components/guides/curl/TestingWithCurl.js]: Or create an issue to fix it.

**Pull request merged**